dns/rfc2136/src/etc/inc/plugins.inc.d/rfc2136.inc

```diff
@@ -200,14 +200,26 @@ function rfc2136_configure_do($verbose = false, $int = '', $updatehost = '', $fo
         if ($need_update) {
             @file_put_contents("/var/etc/nsupdatecmds{$i}", $upinst);
             unset($upinst);
+
             /* invoke nsupdate */
             $cmd = "/usr/local/bin/nsupdate -k {$keyfile}";
             if (isset($dnsupdate['usetcp'])) {
                 $cmd .= " -v";
             }
+
             $cmd .= " /var/etc/nsupdatecmds{$i}";
```

Comment:
Looking at the code, this stuff could really use a rewrite. The current code actually has a race condition if you ask me, the cache files(s) should only be written after an successful update, which currently isn't the case.

A short term solution could be to keep the `mwexec_bg($cmd);` and wrap the update action in a new shell script (`src/bin/nsupdate.sh`), although Ideally one should then also change the `rfc2136_cache_file` function to something easily predictable using the same key. If these cache files are not used anywhere else, it might be an option to calculate one key which is then used for all files (e.g. `{$dnsupdate['interface']}_{$dnsupdate['host']}_{$dnsupdate['server']}{$ipver}`).

The files being `/var/etc/nsupdatecmds{$key}`, `/var/etc/nsupdatekey{$key}`,  `/var/cache/rfc2136_{$key}_4.cache` and `/var/cache/rfc2136_{$key}_6.cache`.

I'm not sure how to feed the complete action to `/usr/sbin/daemon`, but if we push it in a simple script, we only need to pass the new key and the value of `usetcp`, after which we could do something like:

`/usr/local/bin/nsupdate -k ... || rm {both cache files}`

I'm not using this myself (also don't have a test setup), but this is a direction that could work in my opinion.

Reply:
You're right, creating the cache files after the nsupdate was successful would be a better way. I try to implement that in the next week.